doc/protocol.md

```diff
@@ -850,6 +850,8 @@ this payload type:
    refers to a readable directory, an `entries` attribute will be reported
    (described below).  Use `*` to report all files.  Can also be used to
    implement search (`searc*`) or hide temporary files (`[!.]*`).
+ * `targets` (optional string): if set to 'stat' then report information about
+   symlink targets under the `targets` attribute.  See below.
```

Comment:
That is a bit of an outlier. That it uses "stat" is an implementation detail which may even change with the next generation of i/fs/quantumnotify. Do you expect any other mode of this in the future? It could go into `attrs` (perhaps as `target_info` or simply `targets`), or if you don't like that for some reason, it could be a bool?

Reply:
My idea is that we might eventually support `'poll'`/`'scan'` modes or even `'watch'` some day (the way the 'targets' list is structured makes it particularly amenable to periodic scans).  I didn't want to engineer us into a corner where we have a value that's boolean or string.

I didn't put it in `attrs` because it feels more like `entries` (which is also not in `attrs`).  I feel like this could be a better fit in `attrs` indeed, though, particularly since we currently "invent" the need for having this be something other than a boolean (where as with entries, the `fnmatch` thing was a strong motivator).

Totally random aside, something I've been thinking about: what if we renamed `fnmatch` to `entries`?  I think that would be more logical, and get us back to a place where every attribute is named after either an entry in `attrs` or after a option argument, directly.

Reply:
OK, the potential "scan/watch" mode is convincing; I mostly just was curious what motivated this.

I'm torn wrt. renaming fnmatch; `entries="l*"` indeed reflects the output name, but every developer knows what fnmatch is (and often it's not quite clear where you use globs and where regexps). I'm slightly leaning towards fnmatch as it's more descriptive.